applications/StructuralMechanicsApplication/custom_strategies/custom_builder_and_solver/explicit_builder_and_solver.hpp

```diff
@@ -0,0 +1,461 @@
+                            noalias(nodal_inertia) = ZeroVector(3);
+                        }
+
+                    }
+                }
+        }
+    
+        #pragma omp parallel
+        {
+            int k = OpenMPUtils::ThisThread();
+            typename ElementsArrayType::iterator ElemBegin = pElements.begin() + element_partition[k];
+            typename ElementsArrayType::iterator ElemEnd = pElements.begin() + element_partition[k + 1];
+
+            for (typename ElementsArrayType::iterator itElem = ElemBegin; itElem != ElemEnd; itElem++)  
+            {
+                //Getting nodal mass and inertia from element
+                Vector Testtemp;
+
+                // this function needs to be implemented in the respective 
+                // element to provide inertias and nodal masses 
+                itElem->AddExplicitContribution(Testtemp,RESIDUAL_VECTOR,NODAL_INERTIA,rCurrentProcessInfo);
+            }
+        }
+        
+        KRATOS_CATCH( "" )
+    }
+
+    //**************************************************************************
+    //**************************************************************************
+
+    void BuildRHS(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart& rModelPart,
+        TSystemVectorType& b)
+    {
+        KRATOS_TRY
+        
+        // Compute condition contributions to RHS.
+        CalculateAndAddConditionsRHS(pScheme, rModelPart);
+
+        // Compute element contributions to RHS.
+        CalculateAndAddElementsRHS(pScheme, rModelPart);
+
+        
+        KRATOS_CATCH( "" )
+
+    }
+
+
+    
+    
+    //***************************************************************************
+    //***************************************************************************
+
+   
+    void CalculateAndAddConditionsRHS(typename TSchemeType::Pointer pScheme, ModelPart& rModelPart )
+    {
+        KRATOS_TRY
+
+        ProcessInfo& rCurrentProcessInfo      = rModelPart.GetProcessInfo();
+        ConditionsArrayType& pConditions      = rModelPart.Conditions();
+
+        #ifdef _OPENMP
+            int number_of_threads = omp_get_max_threads();
+        #else
+            int number_of_threads = 1;
+        #endif
+
+        vector<unsigned int> condition_partition;
+        OpenMPUtils::CreatePartition(number_of_threads, pConditions.size(), condition_partition);
+
+
+        #pragma omp parallel for 
+        for(int k=0; k<number_of_threads; k++)
+        {
+        typename ConditionsArrayType::ptr_iterator it_begin=pConditions.ptr_begin()+condition_partition[k];
+        typename ConditionsArrayType::ptr_iterator it_end=pConditions.ptr_begin()+condition_partition[k+1];
+
+        for (typename ConditionsArrayType::ptr_iterator it= it_begin; it!=it_end; ++it)
+        {
+
+            LocalSystemVectorType RHS_Condition_Contribution = LocalSystemVectorType(0);
+
+            Element::EquationIdVectorType EquationId; //Dummy
```

Comment:
I would include the dummy statements in the variable name `equation_id_vector_dummy`